framework/src/modules/chain/chain.js

```diff
@@ -570,11 +581,16 @@ module.exports = class Chain {
 		this.blocks.on(EVENT_NEW_BROADHASH, ({ broadhash, height }) => {
 			this.channel.invoke('app:updateApplicationState', { broadhash, height });
 		});
+
+		this.bft.on(EVENT_BLOCK_FINALIZED, ({ height }) => {
+			this.dpos.onBlockFinalized({ height });
+		});
```

Comment:
```js
this.bft.on(EVENT_BLOCK_FINALIZED, this.dpos.onBlockFinalized);
```

Here, since the method is not used like above, and even if it was used like that, I'd still recommend to rename the method to clarify what method is doing. Because event name is already telling which event was triggered. I can't understand what method is doing from `onBlockFinalized` but `updateFinalizedRound` is more clear, I think. The same goes for updateFinishedRound.

What do you think?

Reply:
I agree with Mehmet's idea. I think `onBlockFinalized` is more of an event handler so maybe you could event write it like this: 

`this.bft.on(EVENT_BLOCK_FINALIZED, this.dpos.onBlockFinalized)` and the function implementation could call `updateFinishedRound`. This way if eventually, anything else needs to be performed it can be appended and called.

Or you could directly call `updateFinishedRound`.

Reply:
The reason I am partial towards keeping generic names is to hide the internal logic of the modules from the encapsulating chain module. If we use generic method names, the chain only needs to know that dpos module requires to know when a block is finalized. It doesn't care what is the purpose and how dpos will use this information. For me, this decouples chain module and dpos module implementation such that if in the future, we want to updateFinalizedRound as well as perfom another action `onBlockFinalized` event, the code in chain module will need no update.

Reply:
Since @shu, @limiaspasdaniel and @yatki side with have descriptive names in chain module, please respond to the above point i made, if you still don't agree, i'll change the naming of the event handlers

Reply:
I think your comment is aligned with what I mentioned: 
> this.bft.on(EVENT_BLOCK_FINALIZED, this.dpos.onBlockFinalized) and the function implementation could call updateFinishedRound. This way if eventually, anything else needs to be performed it can be appended and called.

So all good from my side.

Reply:
@SargeKhan I think you make sense :D 👍